spec/integration/post_index_spec.rb

```diff
@@ -0,0 +1,29 @@
+require 'rails_helper'
+
+RSpec.describe 'User post index page', type: :system do
+  let!(:user) do
+    User.create(name: 'John', photo: 'https://example.com/profile.jpg')
+    User.create(name: 'Daniel', photo: 'https://example.com/profile.jpg')
+  end
+
+  let!(:posts) do
+    user.posts.create([
+                        { text: 'First post', comments_counter: 2, likes_counter: 5 },
+                        { text: 'Second post', comments_counter: 3, likes_counter: 7 },
+                        { text: 'Third post', comments_counter: 8, likes_counter: 17 }
+                      ])
+  end
+
+  it 'displays user information and posts' do
+    visit user_posts_path(user)
+
+    expect(page).to have_button('Pagination')
+  end
+
+  it 'redirects to post show page when a post is clicked' do
+    visit user_posts_path(user)
+    expect(page).to have_css('.post-show')
+    expect(page).to have_content('John')
+    expect(page).to have_content("Number of post: #{user.posts_counter}")
+  end
+end
```

Comment:
- [ ] Kindly write the following test scenarios for the `Post#Index` page based on this project's requirements 😎 💻:

* Test that asserts you can see the user's profile picture.
* Test that asserts you can see the user's username.
* Test that asserts you can see the number of posts the user has written.
* Test that asserts you can see a post's title.
* Test that asserts you can see some of the post's body.
* Test that asserts you can see the first comments on a post.
* Test that asserts you can see how many comments a post has.
* Test that asserts you can see how many likes a post has.